### PR TITLE
remove extensions

### PR DESCRIPTION
**Pull request by emschwartz**
oops -- another breaking change. let's wait to bump the version with this one until we have another set of breaking features to add. this removes something that isn't being used right now so it can stay in until the next major release

resolves https://github.com/interledger/js-ilp-core/issues/33

### CODE REVIEW

File: test/data/transfer.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
```
Comment (justmoon) [inline test/data/transfer.json]: This file seems unrelated?

**Comment by justmoon** [comment]
Can you explain why this feature was removed? Do we not need extensibility or is there a better way to do it?

**Comment by emschwartz** [comment]
@justmoon This extension mechanism provides no benefit over instantiating a client and passing it to the Extension constructor. I originally thought this would be a good way to manage extensions, but the main thing I thought would be an extension -- what has become `js-ilp` -- ended up using the functionality of the core client simply by instantiating one, using it, and exposing a different API.

File: test/data/transfer.json (hunk 1 above)
Comment (emschwartz) [inline test/data/transfer.json]: Yup, don't know why that was in there

**Comment by justmoon** [comment]
Makes sense, LGTM
